broker-daemon/broker-rpc/info-service/get-supported-markets.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Gets valid markets from the relayer and populates market information for these markets
+ *
+ * @param {Object} request - request object
+ * @param {Object} request.params
+ * @param {RelayerClient} request.relayer
+ * @param {Logger} request.logger
+ * @param {Engine} request.engines
+ * @param {Map<Orderbook>} request.orderbooks
+
+ * @param {Object} responses
+ * @param {function} responses.EmptyResponse
+ * @return {responses.EmptyResponse}
+ */
+async function getSupportedMarkets ({ params, relayer, logger, engines, orderbooks }, { GetSupportedMarketsResponse }) {
+  const { markets } = await relayer.infoService.getMarkets({})
+
+  let supportedMarkets = []
```

Comment:
[nit] You can use `reduce` here instead of pushing to an array

```
markets.reduce((acc, market) => {
  if (!orderbook.get(market) return acc
  acc.push({.....})
}, [])
```